File: nim/pn.nim
var p: bool = true
let n: int = 100000
echo "2\n3"

for i in countup(5, n, 2):
    for j in countup(3, int(i / 2) + 1, 2):
        if i mod j == 0:
            p = false
            break
        else:
            p = true

    if p:
        echo i
        p = false

File: ocaml/pn.ml
open Printf

let n = 100000

let rec int i j: bool =
    if j > i / 2 then true
    else if i mod j == 0 then false
    else int i (j + 2)

let rec ext i =
    if i > n then ()
    else begin
        if int i 3 then printf "%d\n" i;
        ext (i + 2)
    end

let () =
    printf "2\n3\n";
    ext 5
